sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+            real_IG_value = value
+            size = len(labeled_seg[labeled_seg == value])
+
+    # set all other 0 regions to mark value (1 or 0.5)
+    for value in values:
+        if np.sum(segmented_image[labeled_seg == value]) == 0:
+            if value != real_IG_value:
+                if not mark:
+                    segmented_image_fixed[labeled_seg == value] = 1
+                elif mark:
+                    segmented_image_fixed[labeled_seg == value] = 0.5
+
+    return segmented_image_fixed
+
+
+def mark_faculae(segmented_image, data, res):
+    """
+    Mark faculae seperatley from granules - give them a value of 2 not 1.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect middles.
+    data : `numpy array`
+        The original flux values.
+    res : `float`
+        Spatial resolution (arcsec/pixel) of the data.
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndrray`
+        The segmented image with faculae marked as 1.5.
+    """
+
+    fac_size_limit = 2  # max size of a faculae in sqaure arcsec
+    fac_pix_limit = fac_size_limit / res
+    fac_brightness_limit = np.mean(data) + 0.5 * np.std(data)
+
+    if len(np.unique(segmented_image)) > 3:
+        raise ValueError("segmented_image must have only values of 1, 0, " + "an 0.5 (if dim centers marked)")
+
+    segmented_image = segmented_image
+    segmented_image_fixed = np.copy(segmented_image.astype(float))
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    fac_count = 0
+    for value in values:
+        mask = np.zeros_like(segmented_image)
+        mask[labeled_seg == value] = 1
+        # check that is a 1 (white) region
+        if np.sum(np.multiply(mask, segmented_image)) > 0:
+            region_size = len(segmented_image_fixed[mask == 1])
+            tot_flux = np.sum(data[mask == 1])
+            # check that region is small
+            if region_size < fac_pix_limit:
+                # check that avg flux very high
+                if tot_flux / region_size > fac_brightness_limit:
+                    segmented_image_fixed[mask == 1] = 1.5
+                    fac_count += 1
+
+    gran_count = len(values) - 1 - fac_count  # subtract 1 for IG region
+
+    return segmented_image_fixed, fac_count, gran_count
+
+
+def kmeans_segment(data, llambda_axis=-1):
+    """
+    kmeans clustering: uses a kmeans algorithm to cluster data, in order to
+    independently verify the skimage clustering method (e.g using
+    cross_correlation() below).
+
+    Parameters
+    ----------
+    data : `numpy array`
+        Data to be clustered.
+    llambda_axis : `int`
+        Index for wavelength, -1 if scalar array.
+
+    Returns
+    -------
+    labels : `numpy.ndarray`
+        An array of labels, with 0 = granules, 2 = intergranules,
+        1 = in-between.

```

Comment:
```suggestion
        An array of labels with:
        
        * 0 are granules
        * 2 are intergranules
        * 1 are in-between regions
```
Is the last one I suggested correct?